core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -280,4 +276,12 @@ private[spark] object RDDOperationGraph extends Logging {
     }
     subgraph.append(indent).append("}\n")
   }
+
+  def formatDeterministicLevel(node: RDDOperationNode): String = {
+    node.outputDeterministicLevel match {
+      case DeterministicLevel.INDETERMINATE => " [Indeterminate]"
+      case DeterministicLevel.UNORDERED => " [Unordered]"
+      case _ => ""
```

Comment:
This could be a root cause of future bug. Please follow the existing way by adding back `case DeterministicLevel.DETERMINATE => ""` and raises Exception explicitly for unmatched case.